core/src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -1326,6 +1330,58 @@ protected Engine getEngineOrNull() {
         return this.currentEngineReference.get();
     }
 
+    public void startRecovery(ShardRouting shardRouting, DiscoveryNode localNode, DiscoveryNode sourceNode, RecoveryTargetService recoveryTargetService, RecoveryTargetService.RecoveryListener recoveryListener, RepositoriesService repositoriesService) {
```

Comment:
why do we need to pass the shardRouting here? can't we use the one of this IndexShard?